src/hotspot/share/opto/vectorization.cpp

```diff
@@ -161,11 +174,62 @@ VStatus VLoopAnalyzer::setup_submodules_helper() {
 
   _types.compute_vector_element_type();
 
+  _pointers.compute_and_cache();
+
   _dependency_graph.construct();
 
   return VStatus::make_success();
 }
 
+void VLoopPointers::compute_and_cache() {
+  int number_of_pointers = 0;
+  for (int i = 0; i < _body.body().length(); i++) {
+    MemNode* mem = _body.body().at(i)->isa_Mem();
+    if (mem != nullptr && _vloop.in_bb(mem)) {
+      number_of_pointers++;
+    }
+  }
+
+  uint bytes = number_of_pointers * sizeof(VPointer);
+  _pointers = (VPointer*)_arena->Amalloc(bytes);
```

Comment:
Note: I wish I could use `GrowableArray` here. But I have a `StackObj` that is `NONCOPYABLE`. I thus have to directly construct the `VPointer` into the array, and cannot construct it outside and pass it in. Someday, I hope that `GrowableArray` allows appending with the move-constructor, or something similar.

For now: I simply allocate my own memory, and use the placement-new to construct the `VPointer`s directly into that memory.